Library/ViewModels/SettingsAccountViewModel.swift

```diff
@@ -24,39 +30,61 @@ public protocol SettingsAccountViewModelType {
 public final class SettingsAccountViewModel: SettingsAccountViewModelInputs,
   SettingsAccountViewModelOutputs, SettingsAccountViewModelType {
   public init(_ viewControllerFactory: @escaping (SettingsAccountCellType, Currency) -> UIViewController?) {
-    let userAccountFields = self.viewWillAppearProperty.signal
-      .switchMap { _ in
-        AppEnvironment.current.apiService
-          .fetchGraphUserAccountFields(query: UserQueries.account.query)
-          .materialize()
-      }
+    let userAccountFields = Signal.merge(
+      self.viewDidLoadProperty.signal,
+      self.viewWillAppearProperty.signal.skip(first: 1)
+    )
+    .switchMap { _ in
+      AppEnvironment.current.apiService
+        .fetchGraphUserAccountFields(query: UserQueries.account.query)
+        .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+        .materialize()
+    }
 
     self.fetchAccountFieldsError = userAccountFields.errors().ignoreValues()
 
-    let shouldHideEmailWarning = userAccountFields.values()
-      .map { response -> Bool in
-        guard let isEmailVerified = response.me.isEmailVerified,
-          let isDeliverable = response.me.isDeliverable else {
+    let user = userAccountFields.values().map(\.me)
+
+    let shouldHideEmailWarning = user
+      .map { user -> Bool in
+        if user.isAppleConnected == .some(true) {
+          return true
+        }
+
+        guard let isEmailVerified = user.isEmailVerified,
+          let isDeliverable = user.isDeliverable else {
           return true
         }
 
         return isEmailVerified && isDeliverable
       }
 
-    let shouldHideEmailPasswordSection = userAccountFields.values()
-      .map { $0.me.hasPassword == .some(false) }
+    let shouldHideEmailPasswordSection = user
+      .map { $0.hasPassword == .some(false) || $0.isAppleConnected == .some(true) }
+
+    let shouldShowCreatePasswordFooter = user
+      .map { user -> (Bool, String) in
+        let isAppleConnected = user.isAppleConnected == .some(true)
+        let userHasPassword = user.hasPassword == .some(true)
+        let shouldShow = !userHasPassword && !isAppleConnected
+
+        return (shouldShow, user.email)
+      }
+
+    let isAppleConnectedAccount = user.map { $0.isAppleConnected == .some(true) }
 
-    self.userHasPasswordAndEmailProperty <~ userAccountFields.values()
-      .map { ($0.me.hasPassword == .some(true), $0.me.email) }
+    self.shouldShowCreatePasswordFooterAndEmailProperty <~ shouldShowCreatePasswordFooter
 
-    let chosenCurrency = userAccountFields.values()
-      .map { Currency(rawValue: $0.me.chosenCurrency ?? Currency.USD.rawValue) ?? Currency.USD }
+    let chosenCurrency = user
+      .map { Currency(rawValue: $0.chosenCurrency ?? Currency.USD.rawValue) ?? Currency.USD }
 
-    self.reloadData = Signal.combineLatest(
+    self.reloadData = Signal.zip(
```

Comment:
Sure about this `zip`? 🤔 Can we not get away with a `takeWhen`?

Reply:
Haha this is all covered with tests!

Reply:
😄ok! As long as we're sure everything emits every time! But it looks like it should as everything emits from `user`.